src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java

```diff
@@ -150,23 +152,63 @@ public XSDValidation verifyXSD(MultipartFile xml) {
 
     @Transactional
     public void createIca(@NotNull MultipartFile file, Boolean force) {
-        List<CheckItem> checks = verifyIca(file, force);
+        try{
+            createIca(new ByteArrayInputStream(file.getInputStream().readAllBytes()), force, false);
+        }catch(SAXException | IOException e){
+            throw new AppException(HttpStatus.BAD_REQUEST, ICA_BAD_REQUEST, "Problem when creating new ICA");
+        }
+    }
+
+    @Transactional
+    @java.lang.SuppressWarnings({"javasecurity:S6096", "java:S5443"})
+    public void createMassiveIcas(@NotNull MultipartFile file, Boolean force) {
+        try{
+            ZipInputStream zis = new ZipInputStream(file.getInputStream());
+            ZipEntry zipEntry = zis.getNextEntry();
+            while (zipEntry != null) {
+                File tempFile = File.createTempFile(zipEntry.getName(), "xml");
+                if(!tempFile.isHidden() && !zipEntry.isDirectory()) {
+                    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                    for(int c = zis.read(); c != -1; c = zis.read()){
+                        baos.write(c);
+                    }
+                    createIca(new ByteArrayInputStream(baos.toByteArray()), force, true);
+                }
+                // remove temp file
+                Files.delete(tempFile.toPath());
+
+                //Go to next file inside zip
+                zipEntry = zis.getNextEntry();
+            }
+        } catch(IOException | SAXException e) {
+            throw new AppException(HttpStatus.BAD_REQUEST, ICA_BAD_REQUEST, "Problem when unzipping file");
+        }
+    }
+
+    private void createIca(@NotNull InputStream inputStream, Boolean force, boolean massive) throws SAXException, IOException{
+        List<CheckItem> checks = verifyIca(inputStream, force);
 
         Optional<CheckItem> check = checks.stream()
                 .filter(item -> item.getValid()
                         .equals(CheckItem.Validity.NOT_VALID))
                 .findFirst();
         if (check.isPresent()) {
-            throw new AppException(AppError.ICA_BAD_REQUEST, String.format("[%s] %s", check.get().getValue(), check.get().getNote()));
+            if(!massive) {
```

Comment:
No, if there is at least an error massive upload will be stopped, so there is no reason to add massive boolean.
Besides I think it is better to deny force upload for massive functionality.